src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1215 @@
+instead of belonging to a specific option set (global options are at the top level of the configuration
+hash).
+
+=item line_fmt 
+
+Defines the format used to represent the key/value pair. The following formats are
+supported (see LINE_FORMAT_xxx constants below):
+
+=over
+
+=item *
+
+A SH shell environment variable definition (export key=val).
+
+=item *
+
+A SH shell variable definition (key=val).
+
+=item *
+
+A 'keyword value' line, as used by Apache config files (similar to Config::General).
+
+=item *
+
+A 'setenv keyword=value' line, as used by Xrootd config files mainly. It doesn't work in a CSH shell script (C<=> present).
+
+=item *
+
+A 'set keyword=value' line, as used in a a CSH shell script to define a shell variable.
+
+=back
+
+Inline comments are not supported in 'keyword value' family of formats.
+
+=item value_fmt 
+
+used to indicate how to interpret the configuration value. It is used mainly for
+boolean values, list and hashes. See LINE_VALUE_xxx constants below for the possible values.
+
+=item value_fmt 
+
+used to indicate how to interpret the configuration value. It is used mainly for
+boolean values, list and hashes. See LINE_VALUE_xxx constants below for the possible values.
+
+=back 
+
+An example of rule declaration is:
+
+    my %dpm_config_rules_2 = (
+        "ALLOW_COREDUMP" => "allowCoreDump:dpm;".LINE_FORMAT_SH_VAR.";".LINE_VALUE_BOOLEAN,
+        "GLOBUS_THREAD_MODEL" => "globusThreadModel:dpm;".LINE_FORMAT_ENV_VAR,
+        "DISKFLAGS" =>"DiskFlags:dpm;".LINE_FORMAT_SH_VAR.";".LINE_VALUE_ARRAY,
+       );
+
+For more comprehensive examples of rules, look at L<ncm-dpmlfc> or L<ncm-xrootd> source code in
+configuration-modules-grid repository.
+
+=cut
+
+
+package CAF::RuleBasedEditor;
+
+use strict;
+use warnings;
+use vars qw($EC);
+$EC = LC::Exception::Context->new->will_store_all;
+
+use parent qw(CAF::FileEditor Exporter);
+
+use EDG::WP4::CCM::Element;
+
+use Readonly;
+
+use Encode qw(encode_utf8);
+
+# Constants from FileEditor
+use CAF::FileEditor qw(BEGINNING_OF_FILE ENDING_OF_FILE);
+
+
+=pod
+
+=head2 Rule Constants
+
+The constants described here are used to build the rules. All these
+constants are exported. Add the following to use them:
+
+    use RuleBasedEditor qw(:rule_constants);
+
+There is a different group of constants for each part of the rule.
+
+
+=head3  LINE_FORMAT_xxx: general syntax of the line
+
+=over
+
+=item *
+
+LINE_FORMAT_SH_VAR:         key=val (e.g. SH shell family). A comment is added at the 
+end of the line if it is modified by L<CAF::RuleBasedEditor>.
+
+=item *
+
+LINE_FORMAT_ENV_VAR:        export key=val (e.g. SH shell family). A comment is added at the 
+end of the line if it is modified by L<CAF::RuleBasedEditor>.
+
+
+=item *
+
+LINE_FORMAT_KEY_VAL:        key val (e.g. Xrootd, Apache)
+
+=item *
+LINE_FORMAT_KEY_VAL_SETENV: setenv key=val  (used by Xrootd in particular)
```

Comment:
Good question! In fact, I think we should not multiply the number of format but rather have an option (in VALUE_OPT for example), as mentionned elsewhere, to use space rather than equal as the keyword/value separator. Just that I had no use case yet for the CSH syntax...